### PR TITLE
[ONE PAGE] Added fix for when XHR patch is set to false

### PR DESCRIPTION
**Pull request by yaustar**
A bug was reported for playable ads on Discord for when the option `patch_xhr_out` was set to false for one page exports.

image

Looking at the code, this was because there was a patch for the engine that replaced `Application.configure` function to assume the first argument as a JSON object instead of a URL (this was done for https://github.com/playcanvas/playcanvas-rest-api-tools/issues/35)

When the PlayCanvas Editor changed the build export boilerplate to use `AppBase`, the patch was no longer being applied correctly and we only half fixed it here: https://github.com/playcanvas/playcanvas-rest-api-tools/pull/51 where it only worked if `patch_xhr_out` was true.

This PR now wraps around `http.get` instead of patching `AppBase.configure` as I'm not keen on duplicating logic from the engine (as that can change in the future meaning this patch could get outdated accidentally). Doing it this way makes it a bit more future proof as we are not duplicating any engine code/logic.

### Test plan

- [x] Tested against [Flappy Bird Ad](https://playcanvas.com/project/785157/overview/flappy-bird-for-playable-fb-ad) with `patch_xhr_out` set to false
- [x] Tested against [Flappy Bird Ad](https://playcanvas.com/project/785157/overview/flappy-bird-for-playable-fb-ad) with `patch_xhr_out` set to true

### CODE REVIEW

**Comment by marklundin** [comment]
Thanks for this @yaustar, yep these api tools are particularly vulnerable to changes to the engine/editor. I'd really like to make this more resilient, maybe using some of of scaffolding approach like other FE tools.

File: one-page.js [hunk 1]
```diff
@@ -38,6 +38,13 @@ function inlineAssets(projectPath) {
             };
 
             (function () {
+               // Patch the http get function to check for an object being passed to it and return immediately
```
Comment (marklundin) [inline one-page.js:41]: minor detail, can we align indentation here

Comment (yaustar) [reply one-page.js]: Done!

**Review by marklundin — APPROVED** [review]

**Comment by yaustar** [comment]
> Thanks for this @yaustar, yep these api tools are particularly vulnerable to changes to the engine/editor. I'd really like to make this more resilient, maybe using some of of scaffolding approach like other FE tools.

What do you mean by `scaffolding approach` approach?

**Comment by marklundin** [comment]
> > Thanks for this @yaustar, yep these api tools are particularly vulnerable to changes to the engine/editor. I'd really like to make this more resilient, maybe using some of of scaffolding approach like other FE tools.
> 
> What do you mean by `scaffolding approach` approach?

It's just an idea at this stage, but many libs use a shorthand to 'scaffold' out project boilerplate eg `npm create vue` which essentially creates a folder structure with any relevant libs installed, index template etc. This could be really useful for engine only PC users ie `npm create playcanvas` immediately sets you up with dev server and basic 'Hello World' boilerplate.

As as an extension, you could also have `npm create playcanvas -project 12345` which could do the same but additionally download the project.

**Comment by yaustar** [comment]
> It's just an idea at this stage, but many libs use a shorthand to 'scaffold' out project boilerplate eg `npm create vue` which essentially creates a folder structure with any relevant libs installed, index template etc. This could be really useful for engine only PC users ie `npm create playcanvas` immediately sets you up with dev server and basic 'Hello World' boilerplate.
> 
> As as an extension, you could also have `npm create playcanvas -project 12345` which could do the same but additionally download the project.

Ah, I see! Yes, this was a thought too that it would use it's own boilerplate instead of the published build.

Another thought was to look at ways to port/add the functionality of the engine patches back to the main engine.

**Pull request merged**